packages/app/src/cli/services/deploy/upload.ts

```diff
@@ -241,7 +241,7 @@ async function uploadWasmBlob(extension: FunctionExtension, apiKey: string, toke
   const {url, headers, maxSize} = await getFunctionExtensionUploadURL({apiKey, token})
   headers['Content-Type'] = 'application/wasm'
 
-  const functionContent = await file.read(extension.buildWasmPath())
+  const functionContent = fs.readFileSync(extension.buildWasmPath())
```

Comment:
It looks like you can pass an encoding object here https://github.com/Shopify/cli/blob/main/packages/cli-kit/src/file.ts#L35 which has a default that might be creating the bug. If you pass an empty object here that should take away the encoding passed to `fs.readFile`, which is the same fix I had to do [here](https://github.com/Shopify/cli/commit/f79392179e615a92b8da3f2183b3714575a960bb)